### PR TITLE
Enable new auth providers

### PR DESCRIPTION
**Pull request by Isaiah-Hamilton**
## What kind of change does this PR introduce?

Enable the new auth providers for users to use.

## What is the current behavior?

Missing Notion, LinkedIn, Vonage, and Textlocal.

## What is the new behavior?

Added Notion, LinkedIn, Vonage, and Textlocal.

## Additional context

https://github.com/supabase/infrastructure/pull/4185

### CODE REVIEW

File: web/sidebars.js [hunk 1]
```diff
@@ -134,9 +134,9 @@ module.exports = {
             'guides/auth/auth-spotify',
             'guides/auth/auth-twitch',
             'guides/auth/auth-twitter',
+            'guides/auth/auth-zoom',
```
Comment (kiwicopple) [inline web/sidebars.js:137]: can we sort these alphabetically

**Review by kiwicopple — APPROVED** [review]

File: studio/pages/project/[ref]/auth/settings.tsx [hunk 2]
```diff
@@ -331,7 +341,38 @@ const Settings = () => {
...
+                      showInlineError
+                      errorMessage="Please enter the vonage account sid."
+                    />
+                    <SecretField
+                      name="SMS_TEXTLOCAL_SENDER"
+                      showInlineError
+                      errorMessage="Please enter the vonage auth token."
+                    />
+                  </>
+                )))}
```
Comment (kangmingtay) [inline studio/pages/project/[ref]/auth/settings.tsx:375]: this makes the default phone provider "textlocal", can we leave the default to use twilio instead?

**Review by kangmingtay — CHANGES_REQUESTED** [review]
Hey @Isaiah-Hamilton, you'll also need to add the providers inside the [config file](https://github.com/supabase/supabase/blob/master/studio/pages/api/auth/%5Bref%5D/config.ts#L63-L65) 

You can check that you've added the providers in the right places by searching for all occurrences of `facebook` (for oauth) or `twilio` (for sms) within the studio directory.

Thanks for helping with this 🔥

File: studio/pages/project/[ref]/auth/settings.tsx [hunk 3]
```diff
@@ -415,6 +462,12 @@ const Settings = () => {
             EXTERNAL_GOOGLE_ENABLED: model.EXTERNAL_GOOGLE_ENABLED,
             EXTERNAL_GOOGLE_CLIENT_ID: model.EXTERNAL_GOOGLE_CLIENT_ID || undefined,
             EXTERNAL_GOOGLE_SECRET: model.EXTERNAL_GOOGLE_SECRET || undefined,
+            EXTERNAL_LINKEDIN_ENABLED: model.EXTERNAL_LINKEDIN_ENABLED || undefined,
```
Comment (kangmingtay) [inline studio/pages/project/[ref]/auth/settings.tsx:465]: this shouldn't default to undefined

File: studio/pages/project/[ref]/auth/settings.tsx [hunk 4]
```diff
@@ -415,6 +462,12 @@ const Settings = () => {
...
+            EXTERNAL_LINKEDIN_CLIENT_ID: model.EXTERNAL_LINKEDIN_CLIENT_ID || undefined,
+            EXTERNAL_LINKEDIN_SECRET: model.EXTERNAL_LINKEDIN_SECRET || undefined,
+            EXTERNAL_NOTION_ENABLED: model.EXTERNAL_NOTION_ENABLED || undefined,
```
Comment (kangmingtay) [inline studio/pages/project/[ref]/auth/settings.tsx:468]: likewise, see https://github.com/supabase/supabase/pull/5304/files#r800837298

**Review by kangmingtay — APPROVED** [review]

**Pull request merged**